Tactics/Sym.lean

```diff
@@ -42,25 +44,30 @@ macro "init_next_step" h_run:ident stepi_eq:ident sn:ident : tactic =>
 
 section stepiTac
 
-/-- Apply the relevant pre-generated stepi lemma to a local hypothesis
+/-- Apply the relevant pre-generated stepi lemma to an expression
   `stepi_eq : stepi ?s = ?s'`
-to obtain a new local hypothesis in terms of `w` and `write_mem`
+to add a new local hypothesis in terms of `w` and `write_mem`
   `h_step : ?s' = w _ _ (w _ _ (... ?s))`
 -/
-def stepiTac (stepi_eq h_step : Ident) (ctx : SymContext)
-  : TacticM Unit := withMainContext do
-  let pc := (Nat.toDigits 16 ctx.pc.toNat).asString
-  --  ^^ The PC in hex
-  let step_lemma := mkIdent <| Name.str ctx.program s!"stepi_eq_0x{pc}"
-
-  evalTacticAndTrace <|← `(tactic| (
-    have $h_step :=
-      _root_.Eq.trans (Eq.symm $stepi_eq)
-        ($step_lemma:ident
-          $ctx.h_program_ident:ident
-          $ctx.h_pc_ident:ident
-          $ctx.h_err_ident:ident)
-  ))
+def stepiTac (stepiEq : Expr) (hStep : Name) : SymReaderM Unit := fun ctx =>
+  withMainContext' do
+    let pc := (Nat.toDigits 16 ctx.pc.toNat).asString
+    --  ^^ The PC in hex
+    let stepLemma := Name.str ctx.program s!"stepi_eq_0x{pc}"
+    -- let stepLemma := Expr.const stepLemma []
+
+    let eff := ctx.effects
```

Comment:
No `evalTactic` here? Do we see a perf. improvement now?

Reply:
Indeed, I got rid of the evalTactic so I could use Exprs, and we should see a performance improvement (both from circumventing the evalTactic, and from not having to look up names in the context as often)

Reply:
Thanks, Alex. That'd be awesome!

Reply:
@alexkeizer  Need this branch to lie around then for perf., even after it's merged?

Reply:
If it's merged, I can just use the commit ID from main, so no need